src/test/java/core/basesyntax/FruitShopServiceTest.java

```diff
@@ -0,0 +1,79 @@
+package core.basesyntax;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Storage;
+import core.basesyntax.model.dto.FruitRecordDto;
+import core.basesyntax.service.AdditionStrategy;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.service.FruitShopServiceImpl;
+import core.basesyntax.service.Operation;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.ReduceStrategy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitShopServiceTest {
+    private static List<FruitRecordDto> expectedTransactionDtos;
+    private static FruitShopService fruitService;
+    private static final AdditionStrategy additionStrategy = new AdditionStrategy();
+
+    @After
+    public void clear() {
+        Storage.fruits.clear();
+    }
+
+    @BeforeClass
+    public static void setUp() {
+        Map<Operation, OperationStrategy> operationStrategyMap = new HashMap<>();
+        operationStrategyMap.put(Operation.BALANCE, additionStrategy);
+        operationStrategyMap.put(Operation.SUPPLY, additionStrategy);
+        operationStrategyMap.put(Operation.RETURN, additionStrategy);
+        operationStrategyMap.put(Operation.PURCHASE, new ReduceStrategy());
+
+        expectedTransactionDtos = new ArrayList<>();
+        expectedTransactionDtos.add(new FruitRecordDto(Operation.BALANCE,
```

Comment:
this setup method looks a too big, let's create private static method to fill transactionDtos if you need all of this. Then you just call it here

Reply:
done